src/main/java/org/carlmontrobotics/robotcode2023/subsystems/Roller.java

```diff
@@ -0,0 +1,117 @@
+    {
+        public boolean runsWhenDisabled() {
+            return true;
+        };
+    };
+
+    public Roller() {
+        led.setLength(ledBuffer.getLength());
+        setLedColor(defaultColor);
+        SmartDashboard.putData(this);
+
+        // TODO: Get proper values for the speeds and timings. For future pull requests, do not merge if this is not deleted
+        SmartDashboard.putNumber("Intake Cone Speed", RollerMode.INTAKE_CONE.speed);
+        SmartDashboard.putNumber("Outtake Cone Speed", RollerMode.OUTTAKE_CONE.speed);
+        SmartDashboard.putNumber("Intake Cube Speed", RollerMode.INTAKE_CUBE.speed);
+        SmartDashboard.putNumber("Outtake Cube Speed", RollerMode.OUTTAKE_CUBE.speed);
+        SmartDashboard.putNumber("Intake Cone Time", RollerMode.INTAKE_CONE.time);
+        SmartDashboard.putNumber("Outtake Cone Time", RollerMode.OUTTAKE_CONE.time);
+        SmartDashboard.putNumber("Intake Cube Time", RollerMode.INTAKE_CUBE.time);
+        SmartDashboard.putNumber("Outtake Cube Time", RollerMode.OUTTAKE_CUBE.time);
+        led.start();
+    }
+
+    @Override
+    public void periodic() {
+        
+        SmartDashboard.putBoolean("Has Game Piece", hasGamePiece());
+        RollerMode.INTAKE_CONE.speed = SmartDashboard.getNumber("Intake Cone Speed", RollerMode.INTAKE_CONE.speed);
+        RollerMode.OUTTAKE_CONE.speed = SmartDashboard.getNumber("Outtake Cone Speed", RollerMode.OUTTAKE_CONE.speed);
+        RollerMode.INTAKE_CUBE.speed = SmartDashboard.getNumber("Intake Cube Speed", RollerMode.INTAKE_CUBE.speed);
+        RollerMode.OUTTAKE_CUBE.speed = SmartDashboard.getNumber("Outtake Cube Speed", RollerMode.OUTTAKE_CUBE.speed);
+        RollerMode.INTAKE_CONE.time = SmartDashboard.getNumber("Intake Cone Time", RollerMode.INTAKE_CONE.time);
+        RollerMode.OUTTAKE_CONE.time = SmartDashboard.getNumber("Outtake Cone Time", RollerMode.OUTTAKE_CONE.time);
+        RollerMode.INTAKE_CUBE.time = SmartDashboard.getNumber("Intake Cube Time", RollerMode.INTAKE_CUBE.time);
+        RollerMode.OUTTAKE_CUBE.time = SmartDashboard.getNumber("Outtake Cube Time", RollerMode.OUTTAKE_CUBE.time);
+
+
+        // LED Update
+        {
+            boolean hasGamePiece = hasGamePiece();
+            if (hasGamePiece && !hadGamePiece) {
+                setLedColor(pickupSuccessColor);
+                resetColorCommand.schedule();
+            }
+            hadGamePiece = hasGamePiece;
+        }
+    }
+
+    public void setSpeed(double speed) {
+        motor.set(speed);
+    }
+
+    public void setLedColor(Color color) {
+        for(int i = 0; i < ledBuffer.getLength(); i++)
+            ledBuffer.setRGB(i, color.getRed(), color.getGreen(), color.getBlue());
+        led.setData(ledBuffer);
+    }
+
+    public boolean hasGamePiece() {
+        SmartDashboard.putNumber("dist", Units.metersToInches((distSensor.getRange() - 16)/1000));
+        return Units.metersToInches((distSensor.getRange() - 16)/1000) < 20;
```

Comment:
The distance to the game piece will be valuable to other commands, so it warrants being computed in it's own public method. Also, it should not be computed twice (once for the SmartDashboard and once to be returned).